src/main/java/frc/robot/systems/IntakeFSMSystem.java

```diff
@@ -0,0 +1,707 @@
+				if (input == null) {
+					return IntakeFSMState.MOVE_TO_HOME;
+				}
+
+				if (input.isIntakeButtonPressed() || input.isOuttakeButtonPressed()) {
+					return IntakeFSMState.MOVE_TO_GROUND;
+				} else {
+					if (input.isShootButtonPressed()) {
+						return IntakeFSMState.FEED_SHOOTER;
+					} else {
+						return IntakeFSMState.MOVE_TO_HOME;
+					}
+				}
+
+			case FEED_SHOOTER:
+				if (input.isShootButtonPressed() && !(input.isIntakeButtonPressed()
+					|| input.isOuttakeButtonPressed())) {
+					return IntakeFSMState.FEED_SHOOTER;
+				} else {
+					return IntakeFSMState.MOVE_TO_HOME;
+				}
+
+			case MOVE_TO_GROUND:
+				if (inRange(throughBore.getDistance(), Constants.GROUND_ENCODER_COUNT)) {
+					if (input.isIntakeButtonPressed() && !input.isOuttakeButtonPressed()
+						&& !input.isShootButtonPressed()) {
+						return IntakeFSMState.INTAKING;
+					} else if (input.isOuttakeButtonPressed() && !input.isIntakeButtonPressed()
+								&& !input.isShootButtonPressed()) {
+						return IntakeFSMState.OUTTAKING;
+					}
+				}
+
+				if ((input.isIntakeButtonPressed() || input.isOuttakeButtonPressed())
+					&& !input.isShootButtonPressed()) {
+					return IntakeFSMState.MOVE_TO_GROUND;
+				} else {
+					return IntakeFSMState.MOVE_TO_HOME;
+				}
+
+			case INTAKING:
+				if (inRange(throughBore.getDistance(), Constants.GROUND_ENCODER_COUNT)
+					&& input.isIntakeButtonPressed() && !input.isOuttakeButtonPressed()
+					&& !input.isShootButtonPressed()) {
+					return IntakeFSMState.INTAKING;
+				}
+
+				if (input.isShootButtonPressed()
+					&& !(input.isOuttakeButtonPressed() || input.isIntakeButtonPressed())) {
+					return IntakeFSMState.MOVE_TO_HOME;
+				} else {
+					return IntakeFSMState.MOVE_TO_GROUND;
+				}
+
+			case OUTTAKING:
+				if (inRange(throughBore.getDistance(), Constants.GROUND_ENCODER_COUNT)
+					&& input.isOuttakeButtonPressed() && !input.isIntakeButtonPressed()
+					&& !input.isShootButtonPressed()) {
+					return IntakeFSMState.OUTTAKING;
+				}
+
+				if (input.isShootButtonPressed()
+					&& !(input.isOuttakeButtonPressed() || input.isIntakeButtonPressed())) {
+					return IntakeFSMState.MOVE_TO_HOME;
+				} else {
+					return IntakeFSMState.MOVE_TO_GROUND;
+				}
+
+			default:
+				throw new IllegalStateException("Invalid state: " + currentState.toString());
+		}
+	}
+
+/* ------------------------ Command handlers ------------------------ */
+
+	private boolean inRange(double targetEncoder, double currentEncoder) {
+		return Math.abs(targetEncoder - currentEncoder) < Constants.INRANGE_VALUE;
+	}
+
+	/**
+	 * A PID controller for the pivot motor.
+	 * @param currentEncoderPID The current position of the pivot motor in encoder units.
+	 * @param targetEncoder The target position of the pivot motor in encoder units.
+	 * @return The correct power to send to the pivot motor to achieve the target position.
+	 */
+	private double pid(double currentEncoderPID, double targetEncoder) {
```

Comment:
Well technically this is just proportional control

Reply:
We figured that we wouldn't need to use the I and D so we just ended up going with proportional. I can update the javadoc but would you also want me to rename the method?